### DISCUSSION TITLE
Beamer support?

### DISCUSSION THREAD

**Discussion by kaimikael** [latex3/tagging-project#704] [General]
I saw the list of packages that aren’t yet compatible with tagging, and I presume there is some internal priority even for the ones on the same priority level, but I’d like to put in a vote for support for beamer.  My main document production these days is lecture slides and they are legally required to be accessible.  
(I know, I could work on implementing it myself, but I suspect I’m not at the necessary skill level :-)

**Comment by josephwright** [comment]
Lecture _slides_ are complex - we've had discussions as to whether tagging these is even logical (think for example an animation that uses 5 pages to show one 'slide' - is that one subsection, five subsections, etc.) . Presumably you need to tag a handout version provided to students - which certainly is on the list but is non-trivial.

**Reply by josephwright** [reply]
BTW that's not just a `beamer` point - you have the same question form a GUI presentation program.

**Reply by kaimikael** [reply]
Agreed.  Text-description of complex graphs might not make a lot of sense either, but I guess even just tagging a bullet list would be better than nothing?

**Reply by josephwright** [reply]
Yes but remember PDF is page-based, so we have to provide appropriate tagging for each page or we have to say we'll only tag a 'flattened' version - which is likely the way forward (even if there may still be limitations). How

**Reply by josephwright** [reply]
There's a lot of complexity here: we are not likely to cover `beamer` quickly, and probably not without some source updates - I have been thinking about a blog post on this area. It's certainly on the agenda and I need to think about it for my day job - but that might mean adjusting what is provided as much as tagging what is already there. (Unlike say a journal article, slides are or at least should be largely visual - so tagging may not be that useful, depending on the approach one takes in making slides in the first place.)

**Reply by u-fischer** [reply]
@josephwright  imho your and my slides are typically not so complex and rather easy to tag. If you have a five bullet points that appear on five pages I would simply tag the repetitions on follow up pages as artifact. That is not so different to repeated header of a longtable.

**Reply by kaimikael** [reply]
Right, so maybe a part of the process would be suggestions on how to create slides that even can be accessible.  Now I have to sit down and think a bit :-)

**Reply by josephwright** [reply]
@u-fischer  Perhaps, but we need to cover a wider set of use cases - and I'm not sure that's so easy (e.g. how do we decide on reading order for blocks, columns, ...)

**Reply by u-fischer** [reply]
@josephwright yes I know that there are complicated cases. But that is not different to the rest of the tagging. We should simply start with standard slides ...

**Reply by FrankMittelbach** [reply]
maybe we should even start just with handouts (there the slides are already flattened and if somebody wants to read a presentation handouts view is in my opinion the best usually). Of course some people make complex overlays and then put such decks on the web, but if you think about it: that does kind of work for people who can fast forward on repetitive slides, but seems to me is a nightmare if those are read out to you by some AT processor, e.g., if you get the same slide 5 times each time adding an additional item ...

**Reply by car222222** [reply]
@u-fischer 
Please can you explain what exactly makes a slide "standard".  Is there a published definition?

**Reply by kaimikael** [reply]
Well, PowerPoint seems to think a standard slide consists of a title and five bullet points :-)

**Reply by u-fischer** [reply]
@car222222 standard slides are slides from me and Joseph (and Frank when he doesn't use a mindmap).

**Reply by car222222** [reply]
@FrankMittelbach 
But surely the major purpose of AT is not to "read out" anything!  
But to help people navigate even the most complex material, which may include seeing or providing textual, and other, versions for some of that material.

Or are we only in the business of "text production", rather than the many diverse aspects of useful, widespread accessibility?

**Reply by car222222** [reply]
A very strange name for these!  Why are only these called "standard", and why are they relevant?

**Reply by FrankMittelbach** [reply]
> @FrankMittelbach But surely the major purpose of AT is not to "read out" anything! But to help people navigate even the most complex material, which may include seeing or providing textual, and other, versions for some of that material.

Chris, the problem is that beamer uses a trick to produce visible slides: it generates many pages in which things are added or are removed and they all end up as full pages (repeating data or containing it but invisible, etc). Just look at any beamer dack with say 20 slides shown but in the deck you find 50 or sixty pages. For a person in the audience that can see the presentation it looks as if things got added to the current slide (or removed from it) but in the pdf they are just separate pages and that makes it very difficult to devise a sensible tagging for such a pdf.

One could think of making some of that material artifacts, e.g., ignoring material that was already on the previous page and only tagging new stuff, but none of that is universal if you look at how people use overlays in slides.

This is why I think one has to start with a subset of feature, i.e., a flattened pdf, where each slide is really just a single page and more complex presentation methods geared towards people with vision is out of scope, at least initially.

**Reply by josephwright** [reply]
Handouts definitely the first place to look - I will probably look at the issues in a blog post as I've said. I certainly want _something_ for my own teaching, but that will be for 2025-26 at this point (term is about to start for 2024-25). So in that sense there is a clear priority 'setting' ...

**Reply by car222222** [reply]
@FrankMittelbach 
Yes sure: my observations were not in amy way intended to say amy thing about how to deal with beamer (or indeed with the any other software that can produce/emulate highly dynamic output).

**Reply by FrankMittelbach** [reply]
@car222222 but this discussion topic is about beamer support (and beamer *is* the relevant software when it comes to presentations with LaTeX) so we have to think about what kind of subset of beamer (aka standard slides) can we take on and support initially. Perhaps more general thoughts on highly dynamic output should have its own discussion topic or topics?

**Reply by car222222** [reply]
@FrankMittelbach 
It is not really so restricted by now, but no matter.

My point is primarily about what is the main purpose of tagging, etc, for beamer (or anything): and that is not the production of inappropriate text! 

Apologies for interfering in a "beamer-only" discussion, but note that you will not get very far with useful tagging of beamer without considering these many, wider concerns.

**Reply by car222222** [reply]
Joseph is certainly taking this discussion far wider!

**Comment by davidcarlisle** [comment]
see also https://www.texdev.net/ltx-talk/  and  https://www.texdev.net/ltx-talk/examples for a tagging aware class somewhat inspired by beamer.
